file-formats/dmon/type/zif_aff_dmon_v1.intf.abap

```diff
@@ -0,0 +1,47 @@
+INTERFACE zif_aff_dmon_v1
+  PUBLIC.
+
+  TYPES:
+  "! <p class="shorttext">Caller Program Type</p>
+  "! Caller Program Type
+  "! $values {@link zif_aff_dmon_v1.data:co_caller_program_type}
+    ty_caller_program_type TYPE c LENGTH 4.
+
+  TYPES:
+    "! <p class="shorttext">ABAP Daemon Settings</p>
+    BEGIN OF ty_main,
+      "! $required
+      format_version TYPE zif_aff_types_v1=>ty_format_version,
+      "! <p class="shorttext">Header</p>
+      "! Header
+      "! $required
+      header         TYPE zif_aff_types_v1=>ty_header_60,
+      "! <p class="shorttext">Daemon Class Name</p>
+      "! ABAP Daemon Class Name
+      "! $required
+      daemon_class_name   TYPE zif_aff_types_v1=>ty_object_name_30,
+      "! <p class="shorttext">Caller Program Type</p>
+      "! Caller Program Type
+      "! $required
+      caller_program_type TYPE ty_caller_program_type,
+      "! <p class="shorttext">Caller Program Name</p>
+      "! Caller Program Name
+      "! $required
+      caller_program_name TYPE c LENGTH 40,
```

Comment:
Shall we remove `_program` here?
```suggestion
      "! <p class="shorttext">Caller Type</p>
      "! Caller type
      "! $required
      caller_type TYPE ty_caller_program_type,
      "! <p class="shorttext">Caller Program Name</p>
      "! Caller name
      "! $required
      caller_name TYPE c LENGTH 40,
```

Furthermore, is the object name really 40? Isn't it 30 if I can enter a program, a class or a function group?

Reply:
I talked to Sasan. We want to keep the "_program". We want to specify the caller as a program and not as a person.